src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
```

Comment:
For the second concern, isn't `WorkspaceService` a singleton? You would call `WorkspaceService<SqlToolsSettings>.Instance` to get the instance. No need to pass it as a parameter.